spec/vcr/xendit/report/create/invalid_date_range.yml

```diff
@@ -0,0 +1,67 @@
+---
+http_interactions:
+- request:
+    method: post
+    uri: https://api.xendit.co/reports
+    body:
+      encoding: UTF-8
+      string: '{"type":"BALANCE_HISTORY","filter":{"from":"2021-01-23T04:01:55.574Z","to":"2021-06-23T04:01:55.574Z"}}'
+    headers:
+      Authorization:
+      - Basic eG5kX2RldmVsb3BtZW50X0FPUXJ5b1ExSjVTelNGek1FSjI0c1lxV1VEdWNEZmVLVnpqSWJ0ZWdMWUs3dDFZc0J3NEpNbTlIOUxIdm06
```

Comment:
hmn, api key ini bisa dihapus gak ya ki? kayak yang lain?
Basic <AUTH KEY>

Soalnya ini public repository :)

Reply:
Gw cek di VCR lain, ada beberapa yang ke commit juga, bisa tolong di edit manual juga gak ki? atau better VCRnya kita custom buat automate filter ini:

Kalo dari ChatGPT begini:


To filter out sensitive information like secret keys or API keys from VCR files in RSpec, you can use the `before_record` hook provided by VCR. This hook allows you to modify the recorded HTTP interactions before they are saved to the VCR file.

Here's an example of how you can filter out a secret key from a VCR file:

1. Install the `vcr` gem if you haven't already by adding it to your Gemfile and running `bundle install`:
   ```ruby
   gem 'vcr'
   ```

2. Configure VCR in your RSpec configuration file (e.g., `spec_helper.rb` or `rails_helper.rb`):
   ```ruby
   require 'vcr'

   VCR.configure do |config|
     config.cassette_library_dir = 'spec/vcr_cassettes'
     config.hook_into :webmock
     config.configure_rspec_metadata!
     config.filter_sensitive_data('<API_KEY>') do |interaction|
       # Replace '<API_KEY>' with the sensitive data you want to filter out
       # You can also pass an environment variable here instead of hardcoding the value
       'FILTERED_API_KEY'
     end
     config.before_record do |interaction|
       interaction.filter_sensitive_data('<API_KEY>') do
         'FILTERED_API_KEY'
       end
     end
   end
   ```

   Make sure to replace `'<API_KEY>'` with the actual key you want to filter out. Also, adjust the `cassette_library_dir` to the appropriate location for your VCR cassettes.

3. In your RSpec example, use VCR to record the HTTP interaction as usual:
   ```ruby
   require 'vcr'

   RSpec.describe 'Your example' do
     it 'makes an API request' do
       VCR.use_cassette('api_request') do
         # Make your API request here
       end
     end
   end
   ```

When you run your tests, the recorded interactions will have the sensitive data replaced with `'FILTERED_API_KEY'` in both the request and response. This ensures that the sensitive data is not stored in the VCR file.